src/request/export/ExportFile.js

```diff
@@ -92,71 +153,80 @@ class ExportFile extends Nimiq.Observable {
             return;
         }
 
-        this.setKey(key, this._request.keyInfo.encrypted);
-        this.fire(ExportFile.Events.KEY_CHANGED, {
-            key,
-            isProtected: this._request.keyInfo.encrypted,
-        });
+        this.setKey(key, passwordBuffer);
+        this.fire(ExportFile.Events.KEY_CHANGED, key, passwordBuffer);
+        await this._goToLoginFileDownload();
+
         TopLevelApi.setLoading(false);
     }
 
     /**
-     * used to set the key if already decrypted elsewhere. This will disable the passphrase requirement.
-     * Set to null to reenable passphrase requirement.
-     * @param {Key | null} key
-     * @param {boolean} isProtected
+     *
+     * @param {string} password
      */
-    setKey(key, isProtected) { // eslint-disable-line no-unused-vars
-        this._key = key;
-        if (key) {
-            this._downloadKeyfile.setEncryptedEntropy(new Uint8Array(0), key.deriveAddress('m/44\'/242\'/0\'/0\''));
-            /** @type {HTMLElement} */(document.getElementById(ExportFile.Pages.EXPORT_FILE))
-                .classList.add('show-download');
+    async _setPassword(password) {
+        if (!this._key || !this._key.id) {
+            // this should never happen
+            this._reject(new Errors.KeyguardError('KeyId not set'));
+            return;
         }
+
+        this._key.hasPin = false;
+        this._password = password ? Utf8Tools.stringToUtf8ByteArray(password) : undefined;
+        await KeyStore.instance.put(this._key, this._password);
+
+        const keyInfo = await KeyStore.instance.getInfo(this._key.id);
+        if (!keyInfo) {
+            this._reject(new Errors.KeyNotFoundError());
+            return;
+        }
+        this._request.keyInfo = keyInfo;
```

Comment:
Why get it from the database? What is different, except the `encrypted` part, which could just as well be set directly on the request.keyInfo.

Reply:
`this._request.keyInfo.encrypted` is readonly. Added a setter.